### PR TITLE
EWPP-467: Add footer link sections

### PR DESCRIPTION
**Pull request by ademarco**

### CODE REVIEW

File: src/Entity/FooterLinkSection.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ *   handlers = {
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "list_builder" = "Drupal\oe_corporate_blocks\FooterLinkSectionListBuilder",
+ *     "route_provider" = {
+ *       "html" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider",
+ *     },
+ *     "form" = {
+ *       "add" = "Drupal\oe_corporate_blocks\Form\FooterLinkSectionForm",
+ *       "edit" = "Drupal\oe_corporate_blocks\Form\FooterLinkSectionForm",
+ *       "delete" = "Drupal\oe_corporate_blocks\Form\FooterLinkGeneralDeleteForm"
```
Comment (imanoleguskiza) [inline src/Entity/FooterLinkSection.php]: The problem with using the same cancel form as the links is that on cancelling we will be redirected to the link collection page and not the section collection page.

File: oe_corporate_blocks.post_update.php [hunk 2]
```diff
@@ -109,3 +110,48 @@ function oe_corporate_blocks_post_update_20004(&$sandbox): void {
...
+    $storage->create($config)->save();
+  }
+}
+
+/**
+ * Set "Related sites" as default section for existing general links.
+ */
+function oe_corporate_blocks_post_update_30002(): void {
+  /** @var \Drupal\oe_corporate_blocks\Entity\FooterLinkGeneral[] $general_links */
+  $general_links = \Drupal::entityTypeManager()->getStorage('footer_link_general')->loadMultiple();
```
Comment (imanoleguskiza) [inline oe_corporate_blocks.post_update.php:172]: I am wondering if there is any value on doing this in a batch process.

File: src/Form/FooterLinkSectionForm.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      '#default_value' => $section->id(),
+      '#machine_name' => [
+        'exists' => '\Drupal\oe_corporate_blocks\Entity\FooterLinkSection::load',
+      ],
+      '#disabled' => !$section->isNew(),
+    ];
+
+    $form['weight'] = [
+      '#type' => 'hidden',
+      '#value' => $this->t('weight') ?? 0,
```
Comment (imanoleguskiza) [inline src/Form/FooterLinkSectionForm.php]: I believe this should be $section->getWeight() ?? 0

File: tests/src/Functional/FooterLinkManagementTest.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    $page->pressButton('Save');
+
+    // Assert section is created with default weight.
+    $this->assertLinkSectionEntity('Section 1', 0);
+    $this->drupalGet('/admin/config/footer_link_section');
+
+    // Assert that section is correctly displayed on the overview page.
+    $this->findRowByLabel('Section 1');
+
+    // Assert new weight when saving the overview page.
```
Comment (imanoleguskiza) [inline tests/src/Functional/FooterLinkManagementTest.php]: It feels like we are missing some steps here to change the weight prior to asserting the weight again.

File: tests/src/Traits/AssertFooterLinksTrait.php [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+   * @param string $url
+   *   Link URL.
+   * @param string $section
+   *   Link section name.
+   * @param int $weight
+   *   Link weight.
+   */
+  protected function assertGeneralLinkEntity(string $label, string $url, string $section, int $weight): void {
+    $link = $this->loadLinkByLabel('footer_link_general', $label);
+    $this->assertEqual($link->getUrl()->toString(), $url, "'{$label}' URL was expected to be set to '{$url}', but '{$link->getUrl()->toString()}' was found.");
```
Comment (imanoleguskiza) [inline tests/src/Traits/AssertFooterLinksTrait.php:50]: Usually we use assertEquals where we can put the expected value first.

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

File: src/Entity/FooterLinkSection.php (hunk 1 above)
Comment (ademarco) [reply src/Entity/FooterLinkSection.php]: Yeah, good one, I'll fix this.

File: oe_corporate_blocks.post_update.php (hunk 2 above)
Comment (ademarco) [reply oe_corporate_blocks.post_update.php:172]: Nah, there can't be more than 10-20 links per site anyway, we can handle 100s of entities without any batch.

File: tests/src/Functional/FooterLinkManagementTest.php (hunk 4 above)
Comment (ademarco) [reply tests/src/Functional/FooterLinkManagementTest.php]: Right, it was a leftover from a previous test.

File: tests/src/Traits/AssertFooterLinksTrait.php (hunk 5 above)
Comment (ademarco) [reply tests/src/Traits/AssertFooterLinksTrait.php:50]: The method's signature seems not to agree with that :smile: 
```
  protected function assertEqual($actual, $expected, $message = '') {
```

File: src/Form/FooterLinkSectionForm.php (hunk 3 above)
Comment (ademarco) [reply src/Form/FooterLinkSectionForm.php]: I'm a bad person.

**Review by 22Alexandra — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Pull request merged**